src/Map/TileMap.cpp

```diff
@@ -329,62 +329,40 @@ bool TileMap::tileHighlightVisible() const
 
 void TileMap::draw()
 {
-	Renderer& r = Utility<Renderer>::get();
-
-	int x = 0, y = 0;
-	Tile* tile = nullptr;
+	Renderer& renderer = Utility<Renderer>::get();
 
 	int tsetOffset = mCurrentDepth > 0 ? TILE_HEIGHT : 0;
 
-	for(int row = 0; row < mEdgeLength; row++)
+	for (int row = 0; row < mEdgeLength; row++)
 	{
-		for(int col = 0; col < mEdgeLength; col++)
+		for (int col = 0; col < mEdgeLength; col++)
 		{
-			x = mMapPosition.x() + ((col - row) * TILE_HALF_WIDTH);
-			y = mMapPosition.y() + ((col + row) * TILE_HEIGHT_HALF_ABSOLUTE);
+			Tile& tile = mTileMap[mCurrentDepth][row + mMapViewLocation.y()][col + mMapViewLocation.x()];
 
-			tile = &mTileMap[mCurrentDepth][row + mMapViewLocation.y()][col + mMapViewLocation.x()];
-
-			/// fixme: this is ... well, it's ugly. Find a better way to do this as pretty soon I'm going to need
-			/// an easier way to change tile render color when it comes time to highlight truck routes, comm
-			/// ranges, etc.
-			if(tile->excavated())
+			if (tile.excavated())
 			{
-				if (row == mMapHighlight.y() && col == mMapHighlight.x())
-				{
-					if (mShowConnections && tile->connected())
-					{
-						r.drawSubImage(mTileset, x, y, tile->index() * TILE_WIDTH, tsetOffset, TILE_WIDTH, TILE_HEIGHT, 71, 224, 146, 255);
-					}
-					else
-					{
-						r.drawSubImage(mTileset, x, y, tile->index() * TILE_WIDTH, tsetOffset, TILE_WIDTH, TILE_HEIGHT, 125, 200, 255, 255);
-					}
-				}
-				else
-				{
-					if (mShowConnections && tile->connected())
-					{
-						r.drawSubImage(mTileset, x, y, tile->index() * TILE_WIDTH, tsetOffset, TILE_WIDTH, TILE_HEIGHT, 0, 255, 0, 255);
-					}
-					else
-					{
-						r.drawSubImage(mTileset, x, y, tile->index() * TILE_WIDTH, tsetOffset, TILE_WIDTH, TILE_HEIGHT);
-					}
-				}
+				auto position = mMapPosition + NAS2D::Vector{(col - row) * TILE_HALF_WIDTH, (col + row) * TILE_HEIGHT_HALF_ABSOLUTE};
```

Comment:
I'm not seeing `position` change (this is a rendering function, so it shouldn't) so it probably can be declared `const`.